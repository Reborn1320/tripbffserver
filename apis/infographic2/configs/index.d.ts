export namespace InfographicConfig {
  type Background = {
    backgroundColor?: string;
  };
  type WidthHeight = {
    width?: number;
    height?: number;
  };

  type Infographic = {
    width: number;
    height?: number; //maybe just remove it
    blocks: Block[];
  } & Background &
    ContainerBlock;

  type Block = LocationBlock | BasicBlock;
  type BasicBlock =
    | ContainerBlock
    | ImageBlock
    | TextBlock
    | TripNameTextBlock
    | TripInfoTextBlock
    | LocationImageBlock
    | LocationNameTextBlock
    | LocationFeelingTextBlock
    | LocationHightLightsTextBlock;

  type TripNameTextBlock = {
    type: "text";
    text: "{{trip.name}}";
  } & BaseTextBlock;

  type TripInfoTextBlock = {
    type: "text";
    text: "{{trip.info}}";
  } & BaseTextBlock;

  type LocationBlock = {
    type: "location";
    blocks: BasicBlock[];
    positioning?: Positioning;
  } & WidthHeight;

  type LocationNameTextBlock = {
    type: "text";
    text: "{{location.name}}";
  } & BaseTextBlock;

  type LocationFeelingTextBlock = {
    type: "text";
    text: "{{location.feeling}}";
  } & BaseTextBlock;

  type LocationHightLightsTextBlock = {
    type: "text";
    text: "{{location.hight-lights}}";
  } & BaseTextBlock;

  // todo not sure if this is the correct choice
  type ContainerBlock = {
    type: "container";
    blocks: Block[];
    positioning?: Positioning;
  } & Background & WidthHeight;

  type ImageBlock = {
    type: "image";
    url: string;
    // width: number;
    // height: number;
    positioning: Positioning;
  };

  type BaseTextBlock = {
    fontSize?: string;
    fontFamily?: string;
    color?: string;
    fontWeight?: string;
    textAnchor?: string;
    textTransform?: string;
    positioning: Positioning;
  };

  type TextBlock = {
    type: "text";
    text: string;
    width?: number;
  } & BaseTextBlock;

  type LocationImageBlock = {
    type: "location-image";
    width: number;
    height: number;
    clipPath?: string;
    positioning?: Positioning;
  };

  type Positioning = {
    // width?: number;
    // height?: number;
  } & RelativePosition;
  
  // StackPosition | RelativePosition;

  type StackPosition = {
    width?: number;
    height: number;
  };

  type RelativePosition = {
    top?: number;
    left?: number;
    right?: number;
    bottom?: number;
  };

  // type Bounds = {
  //   top?: number;
  //   bottom?: number;
  //   left?: number;
  //   right?: number;
  //   width?: number;
  //   height: number;
  // }
}
