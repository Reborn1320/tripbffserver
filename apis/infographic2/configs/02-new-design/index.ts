import { config02Locations } from "./2-locations";
import { config03Locations } from "./3-locations";
import { configNLocations } from "./n-locations";
import { config01Location } from "./1-location";

export const fullNewDesignConfig = {
  1: config01Location,
  2: config02Locations,
  3: config03Locations,
  n: configNLocations
};