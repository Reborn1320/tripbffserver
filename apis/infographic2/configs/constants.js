var LOCALE_EN = "en";
var LOCALE_VI = "vi";

var FEELING_LABEL_EN = "Feeling";
var FEELING_LABEL_VI = "Cảm thấy";

var DAY_LABEL_EN = "day";
var DAYS_LABEL_EN = "days";
var DAY_LABEL_VI = "ngày";

var LOCATION_LABEL_EN = "location";
var LOCATIONS_LABEL_EN = "locations";
var LOCATION_LABEL_VI = "địa điểm";

module.exports = {
  FEELING_LABEL_EN,
  FEELING_LABEL_VI,

  DAY_LABEL_EN,
  DAYS_LABEL_EN,
  DAY_LABEL_VI,

  LOCATION_LABEL_EN,
  LOCATIONS_LABEL_EN,
  LOCATION_LABEL_VI,

  LOCALE_EN,
  LOCALE_VI
}