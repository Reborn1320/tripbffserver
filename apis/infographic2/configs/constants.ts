export const LOCALE_EN = "en";
export const LOCALE_VI = "vi";

export const FEELING_LABEL_EN = "Feeling";
export const FEELING_LABEL_VI = "Cảm thấy";

export const DAY_LABEL_EN = "day";
export const DAYS_LABEL_EN = "days";
export const DAY_LABEL_VI = "ngày";

export const LOCATION_LABEL_EN = "location";
export const LOCATIONS_LABEL_EN = "locations";
export const LOCATION_LABEL_VI = "địa điểm";
