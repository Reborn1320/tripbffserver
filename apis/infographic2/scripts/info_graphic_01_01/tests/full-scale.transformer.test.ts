import { InfographicConfig } from "../../../configs";
import { processBlock } from "../transformers";
import {
  config_01_01,
  config_01_02,
  config_01_others,
} from "../../../configs/info_graphic_01_01/config";

describe("complex transformer", () => {
  test("complex configs for 1 location", () => {
    //Arrange
    const config: InfographicConfig.TripInfographic = config_01_01;

    const data = {
      name: " Chuyến đi đầu tiên trên TripBFF",
      fromDate: 1567962000000,
      toDate: 1568221199999,
      locations: [
        {
          locationId: "a10ca9f3-7b62-4899-abae-b13fd9839a5c",
          name: "Phan Xi Păng",
          fromTime: "September 9, 2019",
          toTime: "2019-09-09T08:25:00.000Z",
          feeling: "Tuyệt vời",
          activity: "Đi bộ đường dài ngắn ngày",
          highlights: "Đẹp, Nguy hiểm",
          signedUrl: "http://placekitten.com/700/500",
        },
      ],
      locale: "vi",
      numberOfDays: 1,
    };

    // Act
    const result = processBlock(config, data, { level: 0, location: 0 });

    // Assert
    expect(result).toBeDefined();
    expect(result.cursor.location).toBe(1);
    expect(result.block).toMatchSnapshot();
  });

  test("complex configs for 2 locations", () => {
    //Arrange
    const config: InfographicConfig.TripInfographic = config_01_02;

    const data = {
      name: " Chuyến đi đầu tiên trên TripBFF",
      fromDate: 1567962000000,
      toDate: 1568221199999,
      locations: [
        {
          locationId: "a10ca9f3-7b62-4899-abae-b13fd9839a5c",
          name: "Phan Xi Păng",
          fromTime: "September 9, 2019",
          toTime: "2019-09-09T08:25:00.000Z",
          feeling: "Tuyệt vời",
          activity: "Đi bộ đường dài ngắn ngày",
          highlights: "Đẹp, Nguy hiểm",
          signedUrl: "http://placekitten.com/700/500",
        },
        {
          locationId: "2b877a4c-5bf9-4315-a8b4-02f44971879d",
          name: "Mù Cang Chải",
          fromTime: "September 10, 2019",
          toTime: "2019-09-10T03:16:00.000Z",
          feeling: "Yên bình",
          activity: "Ngắm ruộng lúa",
          highlights: "Đẹp, Rẻ",
          signedUrl: "http://placekitten.com/1200/1200",
        },
      ],
      locale: "vi",
      numberOfDays: 1,
    };

    // Act
    const result = processBlock(config, data, { level: 0, location: 0 });

    // Assert
    expect(result).toBeDefined();
    expect(result.cursor.location).toBe(2);
    expect(result.block).toMatchSnapshot();
  });

  test("complex configs for n locations", () => {
    //Arrange
    const config: InfographicConfig.TripInfographic = config_01_others;

    const data = {
      name: " Chuyến đi đầu tiên trên TripBFF",
      fromDate: 1567962000000,
      toDate: 1568221199999,
      locations: [
        {
          locationId: "a10ca9f3-7b62-4899-abae-b13fd9839a5c",
          name: "Phan Xi Păng",
          fromTime: "September 9, 2019",
          toTime: "2019-09-09T08:25:00.000Z",
          feeling: "Tuyệt vời",
          activity: "Đi bộ đường dài ngắn ngày",
          highlights: "Đẹp, Nguy hiểm",
          signedUrl: "http://placekitten.com/700/500",
        },
        {
          locationId: "2b877a4c-5bf9-4315-a8b4-02f44971879d",
          name: "Mù Cang Chải",
          fromTime: "September 10, 2019",
          toTime: "2019-09-10T03:16:00.000Z",
          feeling: "Yên bình",
          activity: "Ngắm ruộng lúa",
          highlights: "Đẹp, Rẻ",
          signedUrl: "http://placekitten.com/1200/1200",
        },
        {
          locationId: "1f611a70-abde-4e5d-95d3-ae5e7959a9cb",
          name: "Tú Lệ",
          fromTime: "September 10, 2019",
          toTime: "2019-09-10T03:16:00.000Z",
          feeling: "Chill",
          activity: "Ngắm mặt trời lặn",
          highlights: "Đẹp, Đồ ăn ngon",
          // signedUrl: "http://placekitten.com/800/600"
        },
      ],
      locale: "vi",
      numberOfDays: 1,
    };

    // Act
    const result = processBlock(config, data, { level: 0, location: 0 });

    // Assert
    expect(result).toBeDefined();
    expect(result.cursor.location).toBe(3);
    expect(result).toMatchSnapshot();
  });
});
