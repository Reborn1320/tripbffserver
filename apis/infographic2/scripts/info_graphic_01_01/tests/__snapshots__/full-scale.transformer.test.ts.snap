// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex transformer complex configs for 1 location 1`] = `
Object {
  "backgroundColor": "#e3d1a2",
  "blocks": Array [
    Object {
      "height": 1280,
      "type": "image",
      "url": "http://placekitten.com/700/500",
      "width": 1280,
    },
    Object {
      "blocks": Array [
        Object {
          "color": "#d0363b",
          "fontFamily": "Roboto",
          "fontSize": "64px",
          "fontWeight": "bold",
          "positioning": Object {
            "left": 20,
            "top": 20,
          },
          "text": "PHAN XI PĂNG.",
          "textAnchor": "start",
          "textTransform": "uppercase",
          "type": "text",
        },
        Object {
          "color": "#121113",
          "fontFamily": "Roboto",
          "fontSize": "48px",
          "positioning": Object {
            "left": 20,
            "top": 20,
          },
          "text": "Đi bộ đường dài ngắn ngày. Cảm thấy tuyệt vời.",
          "textAnchor": "start",
          "type": "text",
        },
        Object {
          "color": "#121113",
          "fontFamily": "Roboto",
          "fontSize": "48px",
          "positioning": Object {
            "left": 20,
            "top": 20,
          },
          "text": "Đẹp, nguy hiểm.",
          "textAnchor": "start",
          "type": "text",
        },
      ],
      "flex": "column",
      "height": 250,
      "type": "container",
    },
    Object {
      "blocks": Array [
        Object {
          "positioning": Object {
            "bottom": 80,
            "right": 200,
          },
          "type": "image",
          "url": "./data/images/App_Signature.png",
        },
      ],
      "height": 50,
      "type": "container",
    },
  ],
  "flex": "column",
  "height": 1580,
  "type": "container",
  "width": 1280,
}
`;

exports[`complex transformer complex configs for 2 locations 1`] = `
Object {
  "backgroundColor": "rgb(254, 255, 246)",
  "blocks": Array [
    Object {
      "blocks": Array [
        Object {
          "color": "rgb(0, 0, 0)",
          "fontFamily": "Roboto",
          "fontSize": "64px",
          "fontWeight": "bold",
          "positioning": Object {},
          "text": " CHUYẾN ĐI ĐẦU TIÊN TRÊN TRIPBFF",
          "textAnchor": "middle",
          "textTransform": "uppercase",
          "type": "text",
        },
        Object {
          "color": "rgb(0, 0, 0)",
          "fontFamily": "Roboto",
          "fontSize": "64px",
          "fontWeight": "bold",
          "positioning": Object {},
          "text": "1 ngày, 2 địa điểm",
          "textAnchor": "middle",
          "textTransform": "uppercase",
          "type": "text",
        },
      ],
      "flex": "column",
      "height": 170,
      "type": "container",
    },
    Object {
      "backgroundColor": "#fff",
      "blocks": Array [
        Object {
          "blocks": Array [
            Object {
              "height": 840,
              "type": "image",
              "url": "http://placekitten.com/700/500",
              "width": 640,
            },
            Object {
              "color": "#d0363b",
              "fontFamily": "Roboto",
              "fontSize": "64px",
              "fontWeight": "bold",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "PHAN XI PĂNG.",
              "textAnchor": "start",
              "textTransform": "uppercase",
              "type": "text",
            },
            Object {
              "color": "#121113",
              "fontFamily": "Roboto",
              "fontSize": "48px",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "Đi bộ đường dài ngắn ngày. Cảm thấy tuyệt vời.",
              "textAnchor": "start",
              "type": "text",
              "width": 600,
            },
            Object {
              "color": "#121113",
              "fontFamily": "Roboto",
              "fontSize": "48px",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "Đẹp, nguy hiểm.",
              "textAnchor": "start",
              "type": "text",
            },
          ],
          "flex": "column",
          "height": 840,
          "type": "container",
        },
        Object {
          "blocks": Array [
            Object {
              "height": 840,
              "type": "image",
              "url": "http://placekitten.com/1200/1200",
              "width": 640,
            },
            Object {
              "color": "#d0363b",
              "fontFamily": "Roboto",
              "fontSize": "64px",
              "fontWeight": "bold",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "MÙ CANG CHẢI.",
              "textAnchor": "start",
              "textTransform": "uppercase",
              "type": "text",
            },
            Object {
              "color": "#121113",
              "fontFamily": "Roboto",
              "fontSize": "48px",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "Ngắm ruộng lúa. Cảm thấy yên bình.",
              "textAnchor": "start",
              "type": "text",
              "width": 600,
            },
            Object {
              "color": "#121113",
              "fontFamily": "Roboto",
              "fontSize": "48px",
              "positioning": Object {
                "left": 20,
                "top": 60,
              },
              "text": "Đẹp, rẻ.",
              "textAnchor": "start",
              "type": "text",
            },
          ],
          "flex": "column",
          "height": 840,
          "positioning": Object {
            "left": 640,
          },
          "type": "container",
        },
      ],
      "height": 1200,
      "type": "container",
    },
    Object {
      "blocks": Array [
        Object {
          "positioning": Object {
            "bottom": 80,
            "right": 200,
          },
          "type": "image",
          "url": "./data/images/App_Signature.png",
        },
      ],
      "height": 70,
      "type": "container",
    },
  ],
  "flex": "column",
  "height": 1440,
  "type": "container",
  "width": 1280,
}
`;

exports[`complex transformer complex configs for n locations 1`] = `
Object {
  "block": Object {
    "backgroundColor": "#e3d1a2",
    "blocks": Array [
      Object {
        "backgroundColor": "#d0363b",
        "blocks": Array [
          Object {
            "color": "#e3d1a2",
            "fontFamily": "Roboto",
            "fontSize": "84px",
            "fontWeight": "bold",
            "positioning": Object {
              "top": 80,
            },
            "text": " CHUYẾN ĐI ĐẦU TIÊN TRÊN TRIPBFF",
            "textAnchor": "middle",
            "textTransform": "uppercase",
            "type": "text",
          },
          Object {
            "color": "#e3d1a2",
            "fontFamily": "Roboto",
            "fontSize": "74px",
            "positioning": Object {
              "top": 110,
            },
            "text": "1 ngày, 3 địa điểm",
            "textAnchor": "middle",
            "textTransform": "uppercase",
            "type": "text",
          },
        ],
        "flex": "column",
        "height": 370,
        "type": "container",
      },
      Object {
        "blocks": Array [
          Object {
            "blocks": Array [
              Object {
                "strokeColor": "#121113",
                "strokeWidth": 10,
                "type": "line",
                "x1": 1030,
                "x2": 1030,
                "y1": 60,
                "y2": 560,
              },
              Object {
                "fillColor": "red",
                "r": 24,
                "type": "circle",
                "x": 1030,
                "y": 60,
              },
              Object {
                "clipPath": "M269 366.5c-2.2.6-3.9 1.4-5.6 1.6-9.6.8-19.1 1.6-28.7 2.3-4.9.3-9.8.6-14.7.7-5.3.2-10.6.5-15.8.3-5-.1-10-.4-14.8-1.3-6.2-1.1-12.5-1.1-18.6-2.1-15.9-2.6-32.1-1.9-48.1-3-9-.6-18-2-27-3.2-10.2-1.3-20.4-2.8-30.7-4.2-7.5-1-15.1-2-22.6-3-.2 0-.4-.1-.6-.1-6.3-2.7-13-4.8-18.8-8.3-9.8-6-14-15.6-14.4-26.9-.1-3.9-.6-7.7-.6-11.6.1-5.4.5-10.7.8-16 .8-15.8 1.2-31.6 0-47.4-1.1-14.2-2.1-28.3-3.2-42.5-.7-9.1-1.6-18.1-2.2-27.2-.4-6-.3-12.1-.7-18.1-.2-3.4-1.3-6.7-1.4-10.1-.4-8.2-.5-16.5-.7-24.8-.1-3.2-.6-6.4-.5-9.6.1-5 .7-10 .8-15C1 91.2.4 85.4.6 79.7c0-5.5.7-10.9 1.4-16.4.2-1.9 1.6-3.6 2-5.5 1.7-9.4 3.5-18.8 7.9-27.5 1-1.9 2.6-3.6 4.1-5.1.5-.5 1.7-.2 2-.2.7-1.3 1.1-2.6 1.9-3.2 4.3-3.3 8.8-5.7 14.4-6.3 4.2-.5 8.3-2.3 12.5-3.3C51.9 11 57 9.6 62.3 8.9c4.8-.6 9.4-1.9 14.3-2.3 5-.4 10-1.6 15.1-2 13.8-1 27.7-1.7 41.5-2.5 3.4-.2 6.8 0 10.2.1 1.7.1 3.4.5 5.2.7.8.1 1.6-.2 2.4-.2 2.4 0 4.9 0 7.3-.1 4.6-.2 9.3-.7 13.9-.7 4.5 0 9 .3 13.5.5h3.6c5.5-.2 10.9-.2 16.4-.5 5.3-.3 10.5-1 15.8-1.3 2.5-.2 5.1-.2 7.6.3 3.8.7 7.4 1.1 11.2 0 1.4-.4 3.2.3 4.8.5.7.1 1.5.3 1.9 0 2.7-2 5.2-1.5 7.8.2.3.2 1.1-.3 1.7-.3 2.4 0 4.8 0 7.3.1 1.8.1 3.5.7 5.3.9 3.9.5 7.9 1.3 11.9 1.3 7.5.1 15 .4 22.4 2 6.9 1.5 14 1.5 21 2.7 4.2.7 8.5 2 12.1 4.1 3.9 2.2 6.9 5.8 10.4 8.8 1.3 1.1 3.2 1.9 4.1 3.3 2.6 3.8 4.9 7.8 7.2 11.8 1.7 2.9 3.4 5.9 4.7 8.9 1.4 3.5 2.5 7.1 3.6 10.8 1.7 5.8 3.7 11.6 4.8 17.6 1.6 8.4 2.4 17 3.6 25.5.4 2.8.9 5.6 1.1 8.4.6 8 1 15.9 1.5 23.9.7 9.9 1.7 19.7 2.2 29.6.5 9.5.9 19.1.7 28.7-.3 16-1.1 31.9-1.9 47.9-.5 9.9-1.2 19.7-2.2 29.5s-2.5 19.5-3.6 29.3c-.8 6.8-1 13.6-1.9 20.3-.8 6.1-3.8 11.5-7.8 16-3.3 3.8-5.9 8.1-9.3 11.8-3 3.3-6.3 6.6-11.5 6.7-1.2 0-2.3 1.2-3.5 1.8-5.1 2.2-10.4 4.1-15.4 6.5-9.2 4.4-19.1 5.1-29 5.5-3.8.2-7.7-.1-11.5.2-1.9.1-3.7 1-5.5 1.4-.8.2-1.6.1-2.5.2-1.2.1-2.4.4-3.6.3-1.2-.2-2-.6-2.2-.6z",
                "height": 371,
                "positioning": Object {
                  "left": 550,
                  "top": 60,
                },
                "type": "image",
                "url": "http://placekitten.com/700/500",
                "width": 380,
              },
              Object {
                "blocks": Array [
                  Object {
                    "color": "#d0363b",
                    "fontFamily": "Roboto",
                    "fontSize": "64px",
                    "fontWeight": "bold",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "PHAN XI PĂNG.",
                    "textAnchor": "start",
                    "textTransform": "uppercase",
                    "type": "text",
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "Đi bộ đường dài ngắn ngày. Cảm thấy tuyệt vời.",
                    "textAnchor": "start",
                    "type": "text",
                    "width": 600,
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "Đẹp, nguy hiểm.",
                    "textAnchor": "start",
                    "type": "text",
                  },
                ],
                "flex": "column",
                "height": 500,
                "positioning": Object {
                  "left": 1070,
                  "top": 0,
                },
                "type": "container",
                "width": 1030,
              },
            ],
            "height": 500,
            "type": "container",
          },
          Object {
            "blocks": Array [
              Object {
                "strokeColor": "#121113",
                "strokeWidth": 10,
                "type": "line",
                "x1": 1030,
                "x2": 1030,
                "y1": 60,
                "y2": 560,
              },
              Object {
                "fillColor": "red",
                "r": 24,
                "type": "circle",
                "x": 1030,
                "y": 60,
              },
              Object {
                "clipPath": "M269 366.5c-2.2.6-3.9 1.4-5.6 1.6-9.6.8-19.1 1.6-28.7 2.3-4.9.3-9.8.6-14.7.7-5.3.2-10.6.5-15.8.3-5-.1-10-.4-14.8-1.3-6.2-1.1-12.5-1.1-18.6-2.1-15.9-2.6-32.1-1.9-48.1-3-9-.6-18-2-27-3.2-10.2-1.3-20.4-2.8-30.7-4.2-7.5-1-15.1-2-22.6-3-.2 0-.4-.1-.6-.1-6.3-2.7-13-4.8-18.8-8.3-9.8-6-14-15.6-14.4-26.9-.1-3.9-.6-7.7-.6-11.6.1-5.4.5-10.7.8-16 .8-15.8 1.2-31.6 0-47.4-1.1-14.2-2.1-28.3-3.2-42.5-.7-9.1-1.6-18.1-2.2-27.2-.4-6-.3-12.1-.7-18.1-.2-3.4-1.3-6.7-1.4-10.1-.4-8.2-.5-16.5-.7-24.8-.1-3.2-.6-6.4-.5-9.6.1-5 .7-10 .8-15C1 91.2.4 85.4.6 79.7c0-5.5.7-10.9 1.4-16.4.2-1.9 1.6-3.6 2-5.5 1.7-9.4 3.5-18.8 7.9-27.5 1-1.9 2.6-3.6 4.1-5.1.5-.5 1.7-.2 2-.2.7-1.3 1.1-2.6 1.9-3.2 4.3-3.3 8.8-5.7 14.4-6.3 4.2-.5 8.3-2.3 12.5-3.3C51.9 11 57 9.6 62.3 8.9c4.8-.6 9.4-1.9 14.3-2.3 5-.4 10-1.6 15.1-2 13.8-1 27.7-1.7 41.5-2.5 3.4-.2 6.8 0 10.2.1 1.7.1 3.4.5 5.2.7.8.1 1.6-.2 2.4-.2 2.4 0 4.9 0 7.3-.1 4.6-.2 9.3-.7 13.9-.7 4.5 0 9 .3 13.5.5h3.6c5.5-.2 10.9-.2 16.4-.5 5.3-.3 10.5-1 15.8-1.3 2.5-.2 5.1-.2 7.6.3 3.8.7 7.4 1.1 11.2 0 1.4-.4 3.2.3 4.8.5.7.1 1.5.3 1.9 0 2.7-2 5.2-1.5 7.8.2.3.2 1.1-.3 1.7-.3 2.4 0 4.8 0 7.3.1 1.8.1 3.5.7 5.3.9 3.9.5 7.9 1.3 11.9 1.3 7.5.1 15 .4 22.4 2 6.9 1.5 14 1.5 21 2.7 4.2.7 8.5 2 12.1 4.1 3.9 2.2 6.9 5.8 10.4 8.8 1.3 1.1 3.2 1.9 4.1 3.3 2.6 3.8 4.9 7.8 7.2 11.8 1.7 2.9 3.4 5.9 4.7 8.9 1.4 3.5 2.5 7.1 3.6 10.8 1.7 5.8 3.7 11.6 4.8 17.6 1.6 8.4 2.4 17 3.6 25.5.4 2.8.9 5.6 1.1 8.4.6 8 1 15.9 1.5 23.9.7 9.9 1.7 19.7 2.2 29.6.5 9.5.9 19.1.7 28.7-.3 16-1.1 31.9-1.9 47.9-.5 9.9-1.2 19.7-2.2 29.5s-2.5 19.5-3.6 29.3c-.8 6.8-1 13.6-1.9 20.3-.8 6.1-3.8 11.5-7.8 16-3.3 3.8-5.9 8.1-9.3 11.8-3 3.3-6.3 6.6-11.5 6.7-1.2 0-2.3 1.2-3.5 1.8-5.1 2.2-10.4 4.1-15.4 6.5-9.2 4.4-19.1 5.1-29 5.5-3.8.2-7.7-.1-11.5.2-1.9.1-3.7 1-5.5 1.4-.8.2-1.6.1-2.5.2-1.2.1-2.4.4-3.6.3-1.2-.2-2-.6-2.2-.6z",
                "height": 371,
                "positioning": Object {
                  "left": 1150,
                  "top": 60,
                },
                "type": "image",
                "url": "http://placekitten.com/1200/1200",
                "width": 380,
              },
              Object {
                "blocks": Array [
                  Object {
                    "color": "#d0363b",
                    "fontFamily": "Roboto",
                    "fontSize": "64px",
                    "fontWeight": "bold",
                    "positioning": Object {
                      "right": 20,
                      "top": 60,
                    },
                    "text": "MÙ CANG CHẢI.",
                    "textAnchor": "end",
                    "textTransform": "uppercase",
                    "type": "text",
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "right": 20,
                      "top": 60,
                    },
                    "text": "Ngắm ruộng lúa. Cảm thấy yên bình.",
                    "textAnchor": "end",
                    "type": "text",
                    "width": 600,
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "right": 20,
                      "top": 60,
                    },
                    "text": "Đẹp, rẻ.",
                    "textAnchor": "end",
                    "type": "text",
                  },
                ],
                "flex": "column",
                "height": 0,
                "positioning": Object {
                  "left": 0,
                  "top": 0,
                },
                "type": "container",
                "width": 970,
              },
            ],
            "height": 500,
            "type": "container",
          },
          Object {
            "blocks": Array [
              Object {
                "strokeColor": "#121113",
                "strokeWidth": 10,
                "type": "line",
                "x1": 1030,
                "x2": 1030,
                "y1": 60,
                "y2": 560,
              },
              Object {
                "fillColor": "red",
                "r": 24,
                "type": "circle",
                "x": 1030,
                "y": 60,
              },
              Object {
                "clipPath": "M269 366.5c-2.2.6-3.9 1.4-5.6 1.6-9.6.8-19.1 1.6-28.7 2.3-4.9.3-9.8.6-14.7.7-5.3.2-10.6.5-15.8.3-5-.1-10-.4-14.8-1.3-6.2-1.1-12.5-1.1-18.6-2.1-15.9-2.6-32.1-1.9-48.1-3-9-.6-18-2-27-3.2-10.2-1.3-20.4-2.8-30.7-4.2-7.5-1-15.1-2-22.6-3-.2 0-.4-.1-.6-.1-6.3-2.7-13-4.8-18.8-8.3-9.8-6-14-15.6-14.4-26.9-.1-3.9-.6-7.7-.6-11.6.1-5.4.5-10.7.8-16 .8-15.8 1.2-31.6 0-47.4-1.1-14.2-2.1-28.3-3.2-42.5-.7-9.1-1.6-18.1-2.2-27.2-.4-6-.3-12.1-.7-18.1-.2-3.4-1.3-6.7-1.4-10.1-.4-8.2-.5-16.5-.7-24.8-.1-3.2-.6-6.4-.5-9.6.1-5 .7-10 .8-15C1 91.2.4 85.4.6 79.7c0-5.5.7-10.9 1.4-16.4.2-1.9 1.6-3.6 2-5.5 1.7-9.4 3.5-18.8 7.9-27.5 1-1.9 2.6-3.6 4.1-5.1.5-.5 1.7-.2 2-.2.7-1.3 1.1-2.6 1.9-3.2 4.3-3.3 8.8-5.7 14.4-6.3 4.2-.5 8.3-2.3 12.5-3.3C51.9 11 57 9.6 62.3 8.9c4.8-.6 9.4-1.9 14.3-2.3 5-.4 10-1.6 15.1-2 13.8-1 27.7-1.7 41.5-2.5 3.4-.2 6.8 0 10.2.1 1.7.1 3.4.5 5.2.7.8.1 1.6-.2 2.4-.2 2.4 0 4.9 0 7.3-.1 4.6-.2 9.3-.7 13.9-.7 4.5 0 9 .3 13.5.5h3.6c5.5-.2 10.9-.2 16.4-.5 5.3-.3 10.5-1 15.8-1.3 2.5-.2 5.1-.2 7.6.3 3.8.7 7.4 1.1 11.2 0 1.4-.4 3.2.3 4.8.5.7.1 1.5.3 1.9 0 2.7-2 5.2-1.5 7.8.2.3.2 1.1-.3 1.7-.3 2.4 0 4.8 0 7.3.1 1.8.1 3.5.7 5.3.9 3.9.5 7.9 1.3 11.9 1.3 7.5.1 15 .4 22.4 2 6.9 1.5 14 1.5 21 2.7 4.2.7 8.5 2 12.1 4.1 3.9 2.2 6.9 5.8 10.4 8.8 1.3 1.1 3.2 1.9 4.1 3.3 2.6 3.8 4.9 7.8 7.2 11.8 1.7 2.9 3.4 5.9 4.7 8.9 1.4 3.5 2.5 7.1 3.6 10.8 1.7 5.8 3.7 11.6 4.8 17.6 1.6 8.4 2.4 17 3.6 25.5.4 2.8.9 5.6 1.1 8.4.6 8 1 15.9 1.5 23.9.7 9.9 1.7 19.7 2.2 29.6.5 9.5.9 19.1.7 28.7-.3 16-1.1 31.9-1.9 47.9-.5 9.9-1.2 19.7-2.2 29.5s-2.5 19.5-3.6 29.3c-.8 6.8-1 13.6-1.9 20.3-.8 6.1-3.8 11.5-7.8 16-3.3 3.8-5.9 8.1-9.3 11.8-3 3.3-6.3 6.6-11.5 6.7-1.2 0-2.3 1.2-3.5 1.8-5.1 2.2-10.4 4.1-15.4 6.5-9.2 4.4-19.1 5.1-29 5.5-3.8.2-7.7-.1-11.5.2-1.9.1-3.7 1-5.5 1.4-.8.2-1.6.1-2.5.2-1.2.1-2.4.4-3.6.3-1.2-.2-2-.6-2.2-.6z",
                "height": 371,
                "positioning": Object {
                  "left": 550,
                  "top": 60,
                },
                "type": "image",
                "url": "./data/images/EmptyImage01.jpg",
                "width": 380,
              },
              Object {
                "blocks": Array [
                  Object {
                    "color": "#d0363b",
                    "fontFamily": "Roboto",
                    "fontSize": "64px",
                    "fontWeight": "bold",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "TÚ LỆ.",
                    "textAnchor": "start",
                    "textTransform": "uppercase",
                    "type": "text",
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "Ngắm mặt trời lặn. Cảm thấy chill.",
                    "textAnchor": "start",
                    "type": "text",
                    "width": 600,
                  },
                  Object {
                    "color": "#121113",
                    "fontFamily": "Roboto",
                    "fontSize": "48px",
                    "positioning": Object {
                      "left": 20,
                      "top": 60,
                    },
                    "text": "Đẹp, đồ ăn ngon.",
                    "textAnchor": "start",
                    "type": "text",
                  },
                ],
                "flex": "column",
                "height": 500,
                "positioning": Object {
                  "left": 1070,
                  "top": 0,
                },
                "type": "container",
                "width": 1030,
              },
            ],
            "height": 500,
            "type": "container",
          },
        ],
        "flex": "column",
        "height": 1500,
        "type": "container",
      },
      Object {
        "blocks": Array [
          Object {
            "positioning": Object {
              "bottom": 80,
              "right": 200,
            },
            "type": "image",
            "url": "./data/images/App_Signature.png",
          },
        ],
        "height": 200,
        "type": "container",
      },
    ],
    "flex": "column",
    "height": 2070,
    "type": "container",
    "width": 2060,
  },
  "cursor": Object {
    "level": 0,
    "location": 3,
  },
}
`;
